packages/abitype/src/utils.ts

```diff
@@ -397,7 +397,8 @@ export type TypedDataToPrimitiveTypes<
       ? AbiParameterToPrimitiveType<K2, TAbiParameterKind>
       : Error<`Cannot convert unknown type '${K2['type']}' to primitive type.`>
   }
-}
+  // Ensure the result is "Prettied"
+} & unknown
 
 type _TypedDataParametersToAbiParameters<
```

Comment:
Unfortunately I only made it to this point in the file, but you can see how some similar strategies might be applied to these as well as the human-readable types!